### ISSUE TITLE
EPL 1.0 is Deprecated

### ISSUE THREAD

**Issue by dbwiddis** [oshi/oshi#568]
[The EPL-1.0 has been deprecated](https://www.eclipse.org/legal/epl-2.0/faq.php#h.60mjudroo8e5). 

OSHI should update to EPL 2.0 or consider a dual license similar to what JNA has done.

@dblock , you chose the original license.  Any input?

**Comment by dbwiddis** [comment]
Non-blank lines of code git blame assigns to committers, for future reference:

>    1 author Cameron Rollheiser
>    1 author Christian
>    1 author muryoh
>    2 author Nick Sydenham
>    2 author Stefan Reich
>    2 author flake9025
>    3 author Daniel Doubrovkine (dB.) @dblockdotorg
>    3 author Laurent RICHARD
>    3 author dpagano
>    6 author Kald0
>    7 author Mariano A. Amar
>    7 author Rumen Nikiforov
>    8 author Brett Wooldridge
>    8 author Gabor Nagy
>   11 author Marc Miltenberger
>   12 author Pavel Janecka
>   14 author widdisd
>   15 author Tyler Cook
>   23 author Dustin Johnson
>   27 author Detelin
>   31 author S.W
>   37 author Karl Heinz Marbaise
>   43 author Christoph Weitkamp
>   44 author lprimak
>   55 author Christian Kadner
>   65 author Jani Simomaa
>   83 author Darin Howard
>   87 author Izek Greenfield
>  117 author dB
>  132 author bildechinger
>  134 author Chris Browne
>  136 author jpbempel
>  140 author spyhunter99
>  144 author alessiofachechi
>  163 author Ognjen Dragoljevic
>  171 author Corey Holland
>  177 author Tomáš Pavel
>  207 author Michael Desigaud
>  300 author Alessandro Perucchi
>  356 author le-yams
>  443 author lundefugl
>  462 author dblock
>  773 author TzeKei Lee
>  887 author Enrico Bianchi
> 1752 author Jeremy Landis
> 7060 author Daniel Widdis
> 27440 author dbwiddis

**Comment by YoshiEnVerde** [comment]
As far as I can remember:

1. EPL 1.0 to EPL 2.0 migration only requires for the headers to be updated. The very license allows us to just migrate without contributor consent (although, of course, contributor consent will always be a good practice in any FOSS community). This is because the EPL has a section that explicitly states migration between versions as a valid use of the license.

2. EPL 1.0 to EPL 2.0 + GPL is where the problems arise, as enabling the secondary license clause constitutes as re-licensing, and that one requires consensus of **all** contributors.

3. Outside of _supposedly_ fixing some legal loopholes (accrued by the very age of the EPL 1.0), the changes in 2.0 should only impact cases where the secondary license clause is enabled, in which case the code would **always** be licensed under EPL, unless used with a GPL project, wherein it will _convert_ to GPL (it will convert back to EPL 2.0 if the code is removed from the GPL project)

4. For the secondary license clause to be active, we need to explicitly state so in the LICENSE file. If we don't, the clause will remain inactive.

As such:

- As much as a douchebag move it would be, just changing all the headers would be enough to migrate. No contributor agreement needed

- As a more PC move, we could instate a poll for all contributors, put a long deadline, define some way of deciding if the poll came out as Yay or Nay, and move ahead accordingly.

- Under the current license, the code is not under any type of dual licensing, be it by explicitly stating so, or by using the EDL1.0 (instead of EPL1.0) to be able to use BSD Section 3. As such, we won't be any more restricted under EPL2.0 than before. On the contrary, if we enable the secondary license clause (if we want to do so), we'll be able to use OSHI directly in GPL projects (something that was illegal before)

**Comment by dbwiddis** [comment]
Agreed there is no rush to change, and this is the opening round of a conversation on whether we should.  I've spent the last day reading up on what's going on.  If we're happy with the existing license we should at least change from 1 to 2.  However...

One of the main reasons for the change is because EPL is incompatible with GPL.  Under our current license, a new project could not legally incorporate both OSHI and a GPL-licensed library in the same program, and if a new project wished to license under GPL they would need to specify an exception for OSHI.  The secondary license for EPL 2.0 would permit a GPL project (or a project with a GPL dependency) to incorporate OSHI, making it more useful to the community.

@YoshiEnVerde just commented while I was typing this and I'm leaning toward the "poll" version with an intention of unanimity from contributors; with any "no" essentially killing the question, and with a case-by-case evaluation of the contributions of non-respondents.  

That all assumes we want to keep EPL (or EPL2+secondary GPL) rather than, if we're changing the license and asking permissions, revisiting the question of what license we want or should have.  Which is why I'd really like @dblock, the project's originator, to chime in on his reasons for selecting EPL, as I'd like to honor that intent.

**Comment by dblock** [comment]
This was chosen by a lawyer when I was working at AppSecInc. Since then the company had been acquired by Trustwave, which got acquired by Singtel. IANAL, but nobody cares anymore and the community has taken this project very far and the group of maintainers has complete control over the license moving forward.

My personal preference would be to re-license future versions under the MIT license. That said, but the powers vested in me by the Internet I completely delegate this decision to @hazendaz  and @dbwiddis who have done a tremendous job moving OSHI further than I could have ever imagined. Amen.

**Comment by dbwiddis** [comment]
Thanks, @dblock. 

Given the above discussion, I think I'd like to plan to re-license coincident with the 4.0 release (if it ever happens).  I think the 4.0 rewrite will have enough significant new code that it's a natural break from a contribution standpoint.

My preference would be either MIT (for simplicity/brevity) or Apache 2.0 (my personal choice for my other open source contributions), or to exactly match JNA's Apache 2.0/LGPL dual license (because of the significant overlap of our projects).  I don't care too much other than that I'd like OSHI to be useful to the largest number of people (commercial and all other open source including GPL3.)

Even though technically/legally not required, in the interest of the open source community I'll still post it for concurrence in a new issue out of respect for prior contributors and in the interest of transparency to future contributors (especially those putting in effort on 4.0.0).
